include/onnxruntime/core/framework/float16.h

```diff
@@ -23,23 +23,143 @@ namespace onnxruntime {
 
 // MLFloat16
 struct MLFloat16 {
+  // uint16_t values for some special values
+  enum Constants : uint16_t {
+    kSignMask = 0x8000U,
+    kBiasedExponentMask = 0x7C00U,
+    kPositiveInfinityBits = 0x7C00U,
+    kNegativeInfinityBits = 0xFC00U,
+    kPositiveQNaNBits = 0x7E00U,
+    kNegativeQNaNBits = 0xFE00U,
+    kEpsilonBits = 0x4170U,
+    kMinValueBits = 0xFBFFU,
+    kMaxValueBits = 0x7BFFU
+  };
+
   uint16_t val{0};
 
   MLFloat16() = default;
-  explicit constexpr MLFloat16(uint16_t x) : val(x) {}
+  explicit constexpr MLFloat16(uint16_t x) noexcept : val(x) {}
+  constexpr static MLFloat16 FromBits(uint16_t x) noexcept { return MLFloat16(x); }
   explicit MLFloat16(float f);
 
+  static const MLFloat16 NaN;
```

Comment:
constexpr?

Reply:
Not allowed for static instances.

Reply:
My first version of this was constexpr and did not work everywhere.

Reply:
Your example points to a different type than that of the hosting type. Incomplete types are not allowed for `constexpr`.